### ISSUE TITLE
How to update Phpcollab 

### ISSUE THREAD

**Issue by tfederici** [phpcollab/phpcollab#92]
Hi,
we want to update the version of Phpcollab from 2.6.4 to 2.10.3, but we haven't yet found the right way to do that and we're struggling to correctly interpret the instructions contained in \docs\install.md:
***------------------------------------***
### Update (starting from version 1.0) - (deprecated)
- Since 2.2, copy your inc/settings to includes/settings.php
- Save your includes/settings.php and backup your database
- Replace all files in your PhpCollab folder by new files in the last package you download
- Use Update in admin page
1. edit settings (to regenerate new settings.php with eventual new parameters)
2. edit database (to dump new commands according to precedent version)
***------------------------------------***

1) what is meant with "inc/settings" (where is this?) and to where should we copy this before starting?
2) when it is said "Replace all files in your PhpCollab folder" it is meant that: 
    a) should we cancel all existing files in the folder \phpcollab and then copy in it ALL files from the new version's download? and in this case, after that, should we copy the saved folders \files and \logos_clients in the main directory as well as the saved file \includes\settings.php?
    b) or should we intead copy (with replace option) all files of the new version's download onto the existing folder without previously eliminating existing files? 
    c) any different procedure?
3) Anything else we're skipping of the procedure to follow?

We haven't ever got to the point where we have to edit settings or database (we assume after having signed in to the system as administrator from a browser)
We tried more ways (and tentative workarounds), but either the system gives us an error ("...\phpcollab/.htaccess: Invalid command 'php_value', perhaps misspelled or defined by a module not included in the server configuration"), or we get an error page when calling Phpcollab from the browser instead of the page to sign in 

PHP: 7.4
Apache: 2.4
Phpcollab: 2.6.4
present settings: 
# installation type
$installationType = "online"; //select "offline" or "online"
# enable update checker
$updateChecker = "true";

can someone please detail us the steps to follow, or show us what we're doing wrong?
thank in advance!
tommaso

**Comment by mindblender** [comment]
Hi @tfederici 

Did you manually install phpCollab or did you use Scriptaculous?

If it is a manual install, you just need to download the latest release and extract the files to your install location.

As always, I would recommend a backup of all files/folders before hand.

Good luck and let me know if you have any problems or questions.

-Jeff

**Comment by tfederici** [comment]
Hi Jeff,
and thank for your support

actually, we proceeded to a manual install, copying all files of the latest version (2.10.3) over our present phpcollab version 
downloaded 
but when we try to access the system from a browser, instead of the login page we receive an "Internal server error" message
if we go to the Log, we find this error: ""...\phpcollab/.htaccess: Invalid command 'php_value', perhaps misspelled or defined by a module not included in the server configuration"
there is no way to access to the Admin account to inspect and proceed to the update of settings and database as indicated in the instructions
could you spot what should we do in this case, what is the module not included (possibly outside Phpcollab, I think)?
if you need more details, please ask

thank you again!
ciao

**Comment by mindblender** [comment]
Hmm, the `php_value` statements should only turn on features if they are there.  I will do some more digging into them and see if I can make them more stable.

In the mean time, you can replace your .htaccess file with the following and see if it fixes the problem (or maybe it causes more).  All I did was comment out the `php_value` entries.

```
## Deny access to vendor sub directory
RedirectMatch 403 ^/vendor/?$

#php_value error_reporting 32767

#php_value xdebug.var_display_max_depth 5
#php_value xdebug.var_display_max_children 256
#php_value xdebug.var_display_max_data 3000

#    // Display errors
#php_value display_errors 1

#    // Log errors
#php_value log_errors', 1);
#
#    // No error lof message max
#php_value log_errors_max_len 0
#
#    // pecify log file
#php_value error_log ./logs/php_errors.log

#php_value session.use_trans_sid 0

# prevent access to PHP error log
<Files php_errors.log>
 Order allow,deny
 Deny from all
 Satisfy All
</Files>

```

Please let me know if that works for you.

Thanks!

**Comment by tfederici** [comment]
Thank you again Jeff!

actually, the version of the file .htaccess you suggested (with all php_value instructions commented) is nearly the same of the currently working old installation
I thought it was a workaround made by someone who is no more with us and is now unreachable, and I'd like to overcome it

anyway (I hope to be crystal clear, but I know it's a bit convoluted):

if 
we copy all files of the latest version over the existing one (then leaving something untouched, namely \includes\settings.php) and we adopt your version of .htaccess file, 
when we call phpcollab from a browser we get this page
![immagine](https://github.com/phpcollab/phpcollab/assets/31832349/140b3455-e05b-4a04-8e94-4e38cc6e8aee)
and the error in the log is this:
[Thu Jun 20 10:12:41.848631 2024] [core:info] [pid 26924:tid 66180] [client xxx.xx.xx.xx:54469] AH00128: File does not exist: D:/Program Files/Apache2.4/htdocs/collab/public/css/all.min.css, referer ...
actually in \public\css\ there isn't such a file, but there is another one named  fa-all.min.css
strangely, we get the same error page even if we copy this latter file and rename it all.min.css

if instead, 
after having copied all files of the latest version over the existing one and adopted your version of .htaccess file, we remove the file \includes\settings.php 
when we call phpcollab from a browser we get to the setting page (without logging in) 
and when in the step 2 we insert the correct values for database, admin etc. and save we get this error
![immagine](https://github.com/phpcollab/phpcollab/assets/31832349/2f1de0b7-1997-4aaf-96f1-8f0e2ecaf467)

it happens just the same if we cancel all files of the existing installation and we copy all files from the latest version changing .htaccess like you suggested

I really hope to have explained everything

thank you so much again for your support!
tommaso

**Comment by mindblender** [comment]
Hi Tommaso,

Thanks for the additional information.  Looks like the `all.min.css` was a type in the `fatal_error.php` file and probably should be `fa-all.min.css`.  I have not tested that though, but just from looking in the code I think that is what is happening for that error.

As for the the DB error, we would need to see some more details to understand what the error actually was.  Unfortunately, the one displayed is just a general "catch all" error.

Do you see a `~/collab/logs/phpcollab.log` file?  If you look in that file, there should be some errors that might help troubleshoot.  If it exists, you might try doing a `tail -f logs/phpcollab.log` and then reload the page and fill out the form and see what is shown.

Let me know what you find out, and I'm more than willing to help troubleshoot further.

-Jeff

**Comment by tfederici** [comment]
Hi Jeff,
thank you again: hope not to stress you much

try to answer to your points:

_Looks like the all.min.css was a type in the fatal_error.php file and probably should be fa-all.min.css. I have not tested that though, but just from looking in the code I think that is what is happening for that error._
I'm not quite sure to understand here: the typo cannot be only in the /logs/error.php file, because the file fa-all.min.css is actually present in the /public/css folder, and also we get the same error also when we create a copy with the name reported in the error

_As for the the DB error, we would need to see some more details to understand what the error actually was. Unfortunately, the one displayed is just a general "catch all" error._
yes, I understand: is there a log file where we can check to have more details? 

_Do you see a ~/collab/logs/phpcollab.log file? If you look in that file, there should be some errors that might help troubleshoot. If it exists, you might try doing a tail -f logs/phpcollab.log and then reload the page and fill out the form and see what is shown._
actually, there is no \logs folder into the main phpcollab folder (in our case named \collab), nor we can find a phpcollab.log
the logs we usually check (access.log and error.log) are in this folder: /Apache2.4/logs/collab/ 
is there anything else to check, or anything we can do?

thanks again!
ciao
tommaso

**Comment by mindblender** [comment]
> I'm not quite sure to understand here: the typo cannot be only in the /logs/error.php file, because the file fa-all.min.css is actually present in the /public/css folder, and also we get the same error also when we create a copy with the name reported in the error

I did a search in the code for `all.min.css` and the only reference I saw was in the `views/fatal_error.php` file.  That would be causing the `[Thu Jun 20 10:12:41.848631 2024] [core:info] [pid 26924:tid 66180] [client xxx.xx.xx.xx:54469] AH00128: File does not exist: D:/Program Files/Apache2.4/htdocs/collab/public/css/all.min.css` message. however, that would not "break" phpColab, only affect the way things look on that particular page.

> actually, there is no \logs folder into the main phpcollab folder (in our case named \collab), nor we can find a phpcollab.log
the logs we usually check (access.log and error.log) are in this folder: /Apache2.4/logs/collab/
is there anything else to check, or anything we can do?

The errors could also be appearing in the `error.log` file that you mentioned.  I had changed it awhile back to log into `logs/phpcollab.log` file, but if the system can't write to that then it would default back to the general `error.log` file. 

As for tracking down the error, there are a few things you can try to see if it is a problem with phpCollab or something else.

In the phpCollab root folder, create a file called dbtest.php containing the following code:
```
<?php
error_reporting(E_ALL);
ini_set('display_errors',1);

$db_host = 'DB_HOST';
$db_name = 'DB_NAME';
$db_user = 'DB_USER';
$db_password = 'DB_PASSWORD';

try{
    $dbh = new pdo( 'mysql:host='.$db_host.':3306;dbname=' . $db_name,
        $db_user,
        $db_password,
        array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION));
    die(json_encode(array('outcome' => true)));
}
catch(PDOException $PDOException){
    die(json_encode(array('outcome' => false, 'message' => 'Unable to connect', 'error' => $PDOException->getMessage())));
}
catch (Exception $exception) {
    var_dump($exception);
    die(json_encode(array('outcome' => false, 'message' => 'Unable to connect', 'error' => $exception->getMessage())));
}
```

Change the `DB_HOST`, `DB_NAME`, `DB_USER`, and `DB_PASSWORD` to the values you use to connect to your database.  Then access it in the browser `https://YOURSITE/collab/dbtest.php` and see what gets displayed.  It should show something like 
 
<img width="1078" alt="Screen Shot 2024-06-21 at 5 12 53 AM" src="https://github.com/phpcollab/phpcollab/assets/105497/a1bc9e10-54f2-4b37-9c9d-9358c2bdd9bb">

<img width="1008" alt="Screen Shot 2024-06-21 at 5 12 30 AM" src="https://github.com/phpcollab/phpcollab/assets/105497/3599ff3a-e94d-42d2-99c6-b65b99a25c41">

<img width="206" alt="Screen Shot 2024-06-21 at 5 14 22 AM" src="https://github.com/phpcollab/phpcollab/assets/105497/561c35a3-43ac-4abe-a658-9135e123c8e4">


After doing that, if you are successfully able to connect to the DB, i.e. you get the `"outcome": true` message, then there might be something else in phpCollab that is causing the problem and we can take a look at that as needed.

-Jeff

**Comment by tfederici** [comment]
Hi Jeff,
we made the attempt you asked us to make,

first, we copied all files of the latest version over the existing one (but leaving untouched the modified .htaccess in the root, and the file /includes/settings.php), then we called the dbtest.php file from a browser and we got the message:  "outcome": true

after that, we removed the file /includes/settings.php and:
1) when calling the dbtest.php file from a browser we got again the message:  "outcome": true
2) when accessing from a browser phpcollab the installation procedure started:  after having filled and saved the first form with data on database, mail etc., we got the same generic message already documented last time "We had a problem completing the request. Please check and try again. " 

hope everything clear, so that you can suggest further steps

one more thing (just for my curiosity): 
you say that the error [Thu Jun 20 10:12:41.848631 2024] [core:info] [pid 26924:tid 66180] [client xxx.xx.xx.xx:54469] AH00128: File does not exist: D:/Program Files/Apache2.4/htdocs/collab/public/css/all.min.css depends from a typo in the error.php and it shouldn't be fatal
ok, but why we receive such error if we keep our current settings.php file together with a new installation, while we have in the /public/css folder both the file fa-all.min.css and its copy that we named all.min.css ? and why the system is unaccessible and we get the page where it's written "phpcollab appears to have fallen down"?
couldn't it be anything relevant behind such error?

once more, thank you so much !
ciao
tommaso

**Comment by mindblender** [comment]
Hi tommaso,

It sounds like we may be going in circles a bit.

The `docs/install.md` "Update (starting from version 1.0)" instructions are deprecated and should not be followed.  I can understand how the instructions could be found confusing. I will get those cleared up.  My apologies for the confusion.

Let's take a step back and start over and try the following...

**Step 1**:
- restore your backup so your site is up and running/accessible again.

**Step 2**:
- Replace all files (except the `includes/settings.php` file, with the files from the latest release

> **Additional Note:**
When upgrading, you should leave your `includes/settings.php` file in place, otherwise it will trigger a new install.  That being said, there have been a few minor additions that might need to be added to your `includes/settings.php` file depending on how old your original code base is.  Compare your `includes/settings.php` file against the [settings_default.php](https://github.com/phpcollab/phpcollab/blob/master/includes/settings_default.php) in the repo and see if there are any "keys" missing.


Once you have replaced the files from the latest release, check to see if there is a `logs` directory, in your case it should be `collab/logs` and that the web user has write access to it.

Since you pointed out a potential error in the `.htaccess` file, I have pushed an updated version to the repo.  You can view the .htaccess file [here](https://github.com/phpcollab/phpcollab/blob/master/.htaccess), and I would recommend replacing the `.htaccess` file in the release with the one from the repo to rule out any problems there.

Once you have done the above, try accessing the site and see if you still get any errors.  If you do, check the `logs/phpcollab.logs` file and/or the general `error_logs` for PHP errors.  Please let me know if you have any and we can look at addressing them.

-Jeff

**Comment by tfederici** [comment]
Hi Jeff,
we've just followed your detailed instructions and I try to report here the results

working on our currently working installation, we:
- copied and replaced all files from the latest installation
- kept the /includes/settings.php file where we reported the differences from the latest version of settings_default.php (they where not so many however, mostly for comments)
- substituted the .htaccess file in the root with the version you suggested

then, when we called phpcollab from a browser, we 
- got again the already met "Internal server error"
- checked the collab/logs folder where we find on one file (index.php)
- checked the usual error log file (outside, under Apache/logs/collab) where we found the already known error about php_value: ".../Apache2.4/htdocs/collab/.htaccess: Invalid command 'php_value', perhaps misspelled or defined by a module not included in the server configuration, referer: .."

then, for a further test, we commented all the php_value instructions in .htaccess and called again the system from a browser, and we:
- got the usual error page saying "phpcollab have fallen down
- found in error.log this error "AH00128: File does not exist: D:/Program Files/Apache2.4/htdocs/collab/public/css/all.min.css, referer: ..."
- went also on the access.log file in the same folder and there we found (with the same time of the error) this sequence of messages, where you can find a call to all.mini.css:
"GET / HTTP/1.1" 200 498 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"
"GET /javascript/general.js HTTP/1.1" 200 4961 "http://collab..../" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"
"GET /javascript/overlib_mini.js HTTP/1.1" 200 9640 "http://collab.../" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"
"GET /public/css/all.min.css HTTP/1.1" 404 196 "http://collab..." "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"
"GET /public/css/error_page.css HTTP/1.1" 200 474 "http://collab.../" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"
"GET /public/images/robot_fall.png HTTP/1.1" 200 62189 "http://collab.../public/css/error_page.css" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"
"GET /public/images/favicons/favicon-32x32.png HTTP/1.1" 200 2063 "http://collab.../" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36 Edg/126.0.0.0"

finally, just to make a countercheck, we 
- completely removed the current installation
- copied all files from the latest version 
- put the updated version of our settings.php file into /includes
- updated the .htacces file as you said (which we commented internally later)
- and we got in sequence exactly the same errors documented above

I hope to have been clear enough to make the picture helpful, but I know I've been very lengthy and I'm afraid to abuse of your availability
let me know if we can make some more test or intervention

thank you so much
tommaso

**Comment by mindblender** [comment]
Hello tommaso,

Try removing the `.htaccess` file.  I don't see how that would be breaking phpCollab, since it is for Apache.  I am also not seeing anything else in what you have provided that would cause phpCollab to crash.

Let me see if I can recreate your environment and install v2.6 using PHP 7.4 and Apache 2.4.  Then I will try doing the upgrade process.  Unfortunately, it will take a me some time to do that, so perhaps while you wait for me to test that out, you could just restore your previous version so you can keep using phpCollab for now.

I will let you know hat I find out.

-Jeff

**Comment by tfederici** [comment]
Hi Jeff,
we have tried to do what you suggest about .htaccess - copying all files of the version 2.10.3 and then remove the file - and we got the same crash and error about the unexisting all.min.css file that we got when commenting the php_value instructions within it

as for your proposal, let me say that I feel a bit embarassed that you have to do such a job: let me know if and how I can contribute to it
however, I don't think you need to step back to our phpcollab version (2.6.4) that actually works nicely, but for some missing features that should be now in the latest version, and it doesn't break with the .htaccess all commented
actually, the problems reported appear only when you introduce the new version, and even when you introduce it from scratch, as a brand new installation: therefore, I tend to think that, in case, a problem of compativbility is between this version and the rest of the environment (PHP and Apache) 

thanks for all!
ciao
tommaso

**Comment by mindblender** [comment]
Hi Tommaso,

First off, I don't mind helping troubleshoot.  It helps make phpCollab a better product and uncovers problems that I have not encountered.

I still do not see how the missing `all.min.css` file would crash phpCollab.  That file has no functional aspect other than styling, and even then, it is the icons, which are not used on the error page.

I am currently running phpCollab locally in a docker container using PHP 7.4 and Apache 2.2 and MariaDB and unfortunately I have not experienced any major issues that have prevented phpCollab from working.  That is why I want to try and recreate your scenario to see if there is something that I am missing.

Question regarding your "and it doesn't break with the .htaccess all commented" statement.  Are you referring to v2.6 working with the commented out `.htaccess`, or the latest version?

As I mentioned before, I will do some more digging and see if I can recreate your scenario and hopefully find a solution for you.

The only other option I can think of would be to set up a time where we could do a call and share your screen so I can see what is going on and perhaps see some other error.

Sorry my solution(s) are taking so long.  Hopefully I will have better news soon.

-Jeff

**Comment by tfederici** [comment]
Hi Jeff,
I try to answer all points below:

_First off, I don't mind helping troubleshoot.  It helps make phpCollab..._
ok, many thanks (we also somehow contributed a little to the improvement of phpcollab in the past)

_I still do not see how the missing all.min.css..._
actually, this is strange:  maybe the message hides another error

_Question regarding your "and it doesn't break with the .htaccess all commented" ..._
yes, in our working version, in the .htaccess file all instructions are commented, and if we enable them we get the error already documented error ".../Apache2.4/htdocs/collab/.htaccess: Invalid command 'php_value', perhaps misspelled or defined by a module not included in the server configuration, referer: .."
it happens just the same if we introduce the latest version with .htaccess not commented inside
the very difference is that the version 2.6.4 works nicely with .htaccess commented. while the latest version (both if mixed with the existing one or alone) leads to the "famous" error on all.min.css after commenting .htaccess

_The only other option I can think of would be to set up a time where we could do a call and share your screen ..._
I was also thinking to propose that: tell me some possible option to arrange a meeting where I'll share the screen of our server

thanks, ciao
tommaso

**Comment by mindblender** [comment]
Hi Tommaso,

Sorry for the delay in getting back to you.  I tested phpCollab locally using PHP 7.4 and Apache 2.4 and it appeared to work, but I am thinking that it could be something else.

Since it appears that the error(s) appear when you are using the latest `.htaccess` file, let's focus on that.  

Are you using PHP as a CGI/FastCGI?  IF so, then the `php_values` will not work in the `.htaccess` file.

Also, it could be due to an Apache module (like xdebug) not being installed, which could cause Apache to throw the error.

Last, correct me if I am wrong, if you comment out all of the `php_value` directives in the `.htaccess` file, the error(s) go away and you are able to access phpCollab?


I'll wait to see what your response is to the above questions, then we can perhaps see how to proceed.

- Jeff

**Comment by tfederici** [comment]
Hi Jeff,
I'm afraid now things are going a bit far from our present competences...
however, I try to answer your points:

- as far as we can see, we cannot see under Server Manager any _PHP_via_FastCGI_ enabled, but if we go to Task Manager we see a _php-cgi.exe_ process active, and there is a module _mod_proxy_fcgi.so_ present under PHP
- a _xdebug_ module is not present under Apache, but there is one into the same phpcollab folder
- yes, if we comment the _php_value_ directives into _.htaccess_ file, the system works in the version 2.6.4, while it stops immediately after with the "famous" error all.min.css with the version 2.10.3

I hope what above is clear, but I don't know if this can be of any help
maybe, if you have the chance to have a look to our server would be more conclusive, also because we have found several warnings about phpcollab into the php log file, as well as an extensive _collab_php.ini_ with several parameters (mostly set as default) into the PHP folder

let me know
thanks once more
tommaso

**Comment by mindblender** [comment]
Hi Tommaso,

No worries on "a bit far from our present competences",  I am not overly familiar with the workings of Apache + PHP myself.

Are you hosting this on Windows or Linux?

What is confusing me, is that if you were able to run v2.6.4 without any errors/warnings in the PHP log file, and then errors appear once you replace the files from v2.10.3.  The changes made between the versions are mostly logic bugs that were fixed, so that is what I find confusing.

Perhaps try installing version 2.8.2, to see if the issues are still there.  Version 2.9.0 is when I added the `fatal_error.php` screen and introduced the `all.min.css` bug you mention.

Let me know what you find out.

Jeff

**Comment by tfederici** [comment]
Hi Jeff,
I'm reporting my answer here after some days, because I replied to your last message by email, but I'm not fully sure you received that message

so, here you are my replies:
we're hosting PHP under Windows Server
I realize that I may have added confusion:
while searching for possible causes, we found a php.log file with several warnings about phpcollab, but this happens WITH our present version 2.6.4
such version, however, with all php_value instructions commented on .htaccess file, works

we tried also to install, as you suggested, the version 2.8.2, by copying all files over the existing version (just saving the folders \files and \logs_clients and the files setting_php and .htaccess)
here the situation is a bit complicated (hope to be clear):

    when calling the system from a browser, it starts, and we log in
    but after having logged in we receive on the admin page a big message saying: "Attention: Remove the file setup.php!! We can not remove the file, it's not writtable. Please delete manually."
    we cannot find such a setup.php file in the phpcollab folder therefore we go on
    but if navigate the system, everything is there with only one big limitation: there is no icon under Linked content within any Project pages to add, remove etc. files to share
    if we try to add a revision below an existing file, when saving we get to a completely blank page and when moving back we that the revision is not saved

there is also another issue (maybe it might inspire to find a solution), since some time after restoring the backup of the version 2.6.4, we noticed a novelty: even with this version, when we try to upload a new file (or a revision) in an existing project, the operation is regularly registered but the file is not uploaded, and when we check again the linked contents, we find the name, the number, the date and the owner, but beside there is a message "Missing file!" and actually such file is not present in the project related folder under the \Files folder

I hope not to have added further confusion
many thanks as always
ciao
tommaso

**Comment by mindblender** [comment]
Hi Tommaso,

Sorry for the huge delay in getting back to you regarding this.


> I'm reporting my answer here after some days, because I replied to your last message by email, but I'm not fully sure you received that message

No, I did not receive your email reply, wonder where it went.  

> we're hosting PHP under Windows Server

Unfortunately I am unfamiliar with Windows Server.  Are you using IIS or Apache?  In your original post, you mention Apache 2.4.  I just wanted to confirm.

>but after having logged in we receive on the admin page a big message saying: "Attention: Remove the file setup.php!! We can not remove the file, it's not writtable. Please delete manually."
we cannot find such a setup.php file in the phpcollab folder therefore we go on

The `setup.php` file is located at `installation/setup.php`.  The installation process attempts to remove the file, if it does not have permission, or encounters another problem, it displays the message you are seeing.  

> there is no icon under Linked content within any Project pages to add, remove etc.

For clarification, you are referring to these icons?
<img width="224" alt="Screen Shot 2024-08-14 at 10 04 50 AM" src="https://github.com/user-attachments/assets/56c31e5b-4ebc-4323-8736-dddec92cb9f3">

> if we try to add a revision below an existing file, when saving we get to a completely blank page and when moving back we that the revision is not saved

That sounds like there was an issue with either navigating to the page, or when the form is submitted.  A blank page should not appear.  I would need some additional details to help troubleshoot this.


> there is also another issue (maybe it might inspire to find a solution), since some time after restoring the backup of the version 2.6.4, we noticed a novelty: even with this version, when we try to upload a new file (or a revision) in an existing project, the operation is regularly registered but the file is not uploaded, and when we check again the linked contents, we find the name, the number, the date and the owner, but beside there is a message "Missing file!" and actually such file is not present in the project related folder under the \Files folder

I believe I encountered something similar, and started working on a fix for it.  The issue is the DB entry is created before the file is actually uploaded, thus it appears that the file is there when in fact it is not.

Perhaps, as we previously mentioned, the best option for troubleshooting your issue is to do an online meeting so I can see what you are seeing and perhaps offer some additional idea.  If you would like to do that, just let me know what days/times work for you and we can coordinate something.

--Jeff

**Comment by tfederici** [comment]
Hi Jeff, welcome back!
here below the answers to your questions

> Sorry for the huge delay in getting back to you regarding this.

no worries! what is important is that you keep an eye on this

> Unfortunately I am unfamiliar with Windows Server. Are you using IIS or Apache? In your original post, you mention Apache 2.4. I just wanted to confirm.

yes, I confrim Apache 2.4

> The setup.php file is located at installation/setup.php. The installation process attempts to remove the file, if it does not have permission, or encounters another problem, it displays the message you are seeing.

ok, actually the file is there where you say

> For clarification, you are referring to these icons?

yes, just those: they miss only under Contents

> That sounds like there was an issue with either navigating to the page, or when the form is submitted. A blank page should not appear. I would need some additional details to help troubleshoot this.

This happens immediately after clicking on Save the revision (we cannot test this with a new content because of missing icons)

> I believe I encountered something similar, and started working on a fix for it. The issue is the DB entry is created before the file is actually uploaded, thus it appears that the file is there when in fact it is not.

this happens after some of the many update attempts (all failed as you know) and the following restoring of the backup

> Perhaps, as we previously mentioned, the best option for troubleshooting your issue is to do an online meeting so I can see what you are seeing and perhaps offer some additional idea. If you would like to do that, just let me know what days/times work for you and we can coordinate something.

that would be awesome! we cannot use Phpcollab at the moment
I'm in Rome (Italy) but I don't know where you live, thus I don't know wwhat is the best time to you
however, friday 17 at any Rome time between 9 and 17 is fine with me, and the same on monday 19: let me know your best option and also what tool you were thinking to use for sharing my pc


thanks again!
ciao
tommaso

**Comment by mindblender** [comment]
Sounds like there could be multiple issues (pun intended) going on.  

I am in Phoenix, AZ and 9 hours behind you.

Are you familiar with Slack?  If so, join [here](https://join.slack.com/t/phpcollab/shared_invite/zt-94ysxqlt-1Mxttu6iPM~l6gS~qtiEIw) and we can use that to communicate some more and share screens.

I am free this Friday.  Is this correct? [Meeting Planner](https://www.timeanddate.com/worldclock/meetingdetails.html?year=2024&month=8&day=16&hour=17&min=0&sec=0&p1=215&p2=197), I may be available earlier.  If you join Slack, and I become available, I can message you there.

--Jeff

**Comment by tfederici** [comment]
Hi Jeff,
what about meeting tomorrow at 9 (your time, which means 18 Rome time), or is it a bit early to you?
we can even start later (I see you were considering 10) but at 20 (my time) I'll have to leave because of an appointment

as for Slack,, I've an account there (tfederici@tiscali.it), but I used it few times (I didn't even know it was possible to share the screen on it, because I use other tools for this): however, I don't think this would be an issue, thus we can use Slack

let me know about both things
ciao
tommaso,

**Comment by mindblender** [comment]
Hi Tommaso,

I can try and make 9am, my time, work.  

Slack has "huddles" which has screen sharing.  If you have another tool you recommend, let me know.

-Jeff

**Comment by tfederici** [comment]
Time is perfect to me
as for Slack, I don't have a paid subscription that seems necessary to share the screen
therefore, I'd suggest to use the good old Skype, which is fine for talking and sharing the screen at the same time: you don't even need an account to join me, just click this link: https://join.skype.com/FeIKopWQ7IFO
I'll wait for you there!
ciao

**Comment by mindblender** [comment]
Shouldn't need a paid account for Slack to do a huddle.

Either way, I'll install Skype and will visit with you tomorrow.
